src/parser/core/modules/items/bfa/crafted/DarkmoonDeckBlockades.js

```diff
@@ -24,30 +54,139 @@ const DARKMOON_DECK_BLOCKADES_CARDS = [
 
 class DarkmoonDeckBlockades extends Analyzer {
   healing = 0;
+  actualStaminaIncreasePerCard = {};
+  cardTracker = {};
 
   constructor(...args) {
     super(...args);
     this.active = this.selectedCombatant.hasTrinket(ITEMS.DARKMOON_DECK_BLOCKADES.id);
+    if (this.active) {
+      Object.keys(DARKMOON_DECK_BLOCKADES_CARDS).forEach((buffId) => {
+        const defaultCard = DARKMOON_DECK_BLOCKADES_CARDS[buffId];
+        const actualStaminaIncrease = calculatePrimaryStat(355, defaultCard.staminaIncrease, this.selectedCombatant.getItem(ITEMS.DARKMOON_DECK_BLOCKADES.id).itemLevel);
+        this.actualStaminaIncreasePerCard[buffId] = actualStaminaIncrease;
+      });
+    }
+  }
+
+  _isBlockadesCard(spellId) {
+    const cardId = '' + spellId;
+    const darkmoonSpells = Object.keys(DARKMOON_DECK_BLOCKADES_CARDS);
+    return darkmoonSpells.includes(cardId);
   }
 
   on_byPlayer_heal(event) {
-    const spellId = event.ability.guid;
-    if (!DARKMOON_DECK_BLOCKADES_CARDS.includes(spellId)) {
+    if (!this._isBlockadesCard(event.ability.guid)) {
       return;
     }
     this.healing += (event.amount || 0) + (event.absorbed || 0);
   }
 
-  item() {
+  on_toPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (!this._isBlockadesCard(spellId)) {
+      return;
+    }
+    if (!this.cardTracker.hasOwnProperty(spellId)) {
+      this.cardTracker[spellId] = [];
+    }
+    this.cardTracker[spellId].push({
+      start: event.timestamp,
+      end: this.owner.fight.end_time,
+    });
+  }
+
+  on_toPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if (!this._isBlockadesCard(spellId)) {
+      return;
+    }
+    if (!this.cardTracker.hasOwnProperty(spellId)) {
+      return;
+    }
+    const lastOccurrenceOfThisCard = this.cardTracker[spellId].pop();
```

Comment:
Is there a reason why you're poping and pushing the last item instead of just accessing it via the last index to update the `end`-prop?